### ISSUE TITLE
Oddities with finding references

### ISSUE THREAD

**Issue by nemethf** [joaotavora/eglot#227]
I append xref-find-references to xref-prompt-for-identifier, therefore it doesn't prompt for identifier.  However, today I tried to take advantage of the prompting feature and the result was a bit unexpected.  Take the following php file and the [php-language-server](https://github.com/felixfbecker/php-language-server/).

```php
<?php

function func1() {
    return;
}
function func2() {
    return func1();
}

$var1 = func1();
$var2 = func2($var1);
$var3 = func2($var1);
?>
```

And type the following:
```
M-< C-u M-? func1 RET --> failure
M-< C-u M-? TAB func1 RET --> OK

M-< C-u M-? TAB --> no $var1
C-s var1 RET C-u M-? RET --> OK
```
I think the first line should succeed just like the second one.  I suppose the culprit is that xref-backend-references of eglot does not refresh eglot--xref-known-symbol in the first case.

The third line shows that $var1 is not available for completion, yet it is offered as a default and the lookup is successful in the forth line.  I don't understand this, but my guess is that eglot doesn't ask the server for the variables when it sets eglot--xref-known-symbol.

Is it worth the trouble to fix these?  I don't know.  But I've also found a typo in xref-backend-references that is easy to fix: a "t" is missing in "Don' know where ..."

Thanks.

**Comment by joaotavora** [comment]
Yes this is mostly broken.  I was trying to fix it in  #147 but got sidetracked.  Would you like to tackle it?  If so I can hook you up with my prior art and/or Stefan Monnier who has some ideas on how to do it.

**Comment by nemethf** [comment]
I've read that issue and the relevant part of the protocol specification. It's an interesting problem and I have some ideas how to handle it, but I have quite a few things in my queue and I would almost never use this feature.  Nevertheless, I would try to implement something in the forthcoming ... months.

**Comment by joaotavora** [comment]
Closing this, it was fixed, in #314
